lib/ansible/modules/windows/win_disk_facts.py

```diff
@@ -0,0 +1,600 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2017, Marc Tschapek <marc.tschapek@itelligence.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = r'''
+---
+module: win_disk_facts
+version_added: '2.5'
+short_description: Show the attached disks and disk information of the target host
+description:
+   - With the module you can retrieve and output detailed information about the attached disks of the target and
+     it's volumes and partitions if existent.
+requirements:
+    - Windows 8.1 / Windows 2012R2 (NT 6.3) / PowerShell 4.0 or newer
```

Comment:
I would have thought the only restriction would be Windows 8.1 / Windows Server 2012 (NT 6.2), it works when I try it locally.

Reply:
Could you please test this in Win2012 with the virtual disk properties (you would need a Stroage Space and create a virtual disk there)?
Because I am not sure whether all virtual disk properties were available in Win2012.

Reply:
One additional thing:
I have added a check now which OS version is used:
$([System.Environment]::OSVersion.Version.Major).$([System.Environment]::OSVersion.Version.Minor)" -ge 6.3

But would it not be better to check against the PowerShell version?

Reply:
2012 comes with Powershell 3 which seems to work fine with the code so far so the restriction isn't there. I can't test with storage spaces but that should not be a reason why we would restrict the version.